MPChartLib/src/main/java/com/github/mikephil/charting/components/YAxis.java

```diff
@@ -357,12 +378,54 @@ public boolean needsOffset() {
             return false;
     }
 
+    /**
+     * Sets min value restriction for autoscale
+     */
+    public void setAutoScaleMinRestriction(float restrictionValue) {
+        mUseAutoScaleRestrictionMin = true;
+        mAutoScaleMinRestriction = restrictionValue;
+    }
+
+    /**
+     * Sets max value restriction for autoscale
+     */
+    public void setAutoScaleMaxRestriction(float restrictionValue) {
+        mUseAutoScaleRestrictionMax = true;
+        mAutoScaleMaxRestriction = restrictionValue;
+    }
+
+    /**
+     * Resets min value restriction for autoscale
+     */
+    public void resetAutoScaleMinRestriction() {
```

Comment:
replace both `resetAutoScaleMinRestriction()` and `resetAutoScaleMaxRestriction()` with appropriate `setUseAutoScale???Restriction()` and `isUseAutoScale???Restriction()` getters and setters.